appservice/src/deploy/localGitDeploy.ts

```diff
@@ -30,10 +30,13 @@ export async function localGitDeploy(client: SiteClient, fsPath: string): Promis
             try {
                 const status: git.StatusResult = await localGit.status();
                 if (status.files.length > 0) {
+                    telemetryProperties.pushWithUncommitChanges = 'false';
```

Comment:
I'd suggest using `cancelStep`. It's a somewhat common pattern that we already use and it's already picked up in a few queries I think. Here's an example: https://github.com/Microsoft/vscode-azureappservice/blob/80fa5b5317389e5549b86a2bc038808666cf806d/src/commands/deploy.ts#L136

I would do this before the dialog:
```
telemetryProperties.cancelStep = 'pushWithUncommitChanges';
```

And this after the dialog:
```
telemetryProperties.cancelStep = undefined;
telemetryProperties.pushWithUncommitChanges = 'true';
```